wynnset/tapestry-wp
# 701 - Improve Docker configuration

Closes #701 

## What?

This PR completely overhauls the current docker setup to have:

- **Improved debugging** by making Cypress no longer a docker container but rather part of the app
- **Automated installation** so setting up Github actions is easier and on-boarding new hires is faster

## Rationale

- Cypress needed to be extracted out of docker for easier test debugging and coverage reports
- PR #645 has a lot of "hacky" workarounds to make WP install programmatically. This PR improves the docker config to come bundled with WP-CLI so that installation can be done through a bash script.
- I remember encountering a lot of setup issues when Cody and Sean was on-boarding. This PR also automates setup for the local dev environment so that the new hires (hopefully) won't run into much problems.

## Testing

- In the root directory, run `make init`. This will spin up the docker containers, configure WP to the settings that Tapestry requires, and install the required dependencies for the Vue app.
  - This will create a new directory `data` that contains two sub-directories `wp` and `db`. These directories contain the files for the wordpress installation and the wordpress db respectively.
  - If you get an error message along the lines of "Not a wordpress installation", go to the `Makefile`, change `@sleep 15` to a higher number and run `make init` again.
- Once everything's installed, run `make start` to start the development server.
- Visit `localhost:8000` in your browser. If everything works you should see a webpage with the title "Tapestry".
- Go to `/wp-admin` and login with the admin user. The username is `admin` and the password is `secret`.
- Double check that everything works. Go to plugins and check that both H5P and Tapestry is installed and activated.
- Make a new Tapestry and ensure you don't run into any issues there.
- To shut down the server, open a new terminal window in the `tapestry-wp` directory and run `make stop`.
